### PR TITLE
Add NCryptFreeObject

### PR DESCRIPTION
**Pull request by sfackler**

### CODE REVIEW

**Comment by retep998** [comment]
Don't forget to add ncrypt to the dev dependencies of winapi so CI can actually test building it.

**Comment by sfackler** [comment]
Fixed!

**Comment by sfackler** [comment]
Ping

**Comment by retep998** [comment]
This PR will need to be updated to the winapi 0.3 new world order. https://github.com/retep998/winapi-rs/issues/316

**Comment by sfackler** [comment]
Updated! The mingw dlls have not been added anywhere since the appropriate crates haven't been created yet.

**Comment by retep998** [comment]
Could you rebase once a nightly comes out with https://github.com/rust-lang/rust/pull/37361 so this PR can be tested properly?

**Comment by sfackler** [comment]
Sure, will do.

**Comment by sfackler** [comment]
Updated, but looks like the GNU build is failing due to the missing library shims.

File: src/um/mod.rs [hunk 1]
```diff
@@ -14,3 +14,4 @@
 #[cfg(feature = "um-wincon")] pub mod wincon;
 #[cfg(feature = "um-wingdi")] pub mod wingdi;
 #[cfg(feature = "um-winnt")] pub mod winnt;
+#[cfg(feature = "um-ncrypt")] pub mod ncrypt;
```
Comment (retep998) [inline src/um/mod.rs]: Alphabetical order

File: Cargo.toml [hunk 2]
```diff
@@ -69,7 +73,9 @@ headers = ["headers-shared", "headers-um", "headers-vc"]
 "vc-vcruntime" = []
 libraries = [
     "kernel32",
-    "ole32"
+    "ole32",
+    "ncrypt"
```
Comment (retep998) [inline Cargo.toml]: Alphabetical order

File: Cargo.toml [hunk 3]
```diff
@@ -56,6 +59,7 @@ headers = ["headers-shared", "headers-um", "headers-vc"]
 "um-combaseapi" = ["shared-basetsd", "shared-minwindef", "shared-wtypesbase", "um-objidlbase", "ole32"]
 "um-consoleapi" = ["shared-minwindef", "um-wincon", "um-winnt", "kernel32"]
 "um-minwinbase" = ["shared-ntstatus"]
+"um-ncrypt" = ["shared-basetsd", "shared-sspi"]
```
Comment (retep998) [inline Cargo.toml]: This should also depend on `"ncrypt"` because it has functions from that library.

**Comment by sfackler** [comment]
Updated!

**Comment by retep998** [comment]
I added crates in the `i686` and `x86_64` folders to provide mingw import libraries. Could you put the ncrypt import libraries in there?

File: Cargo.toml (hunk 2 above)
Comment (retep998) [inline Cargo.toml]: I think you might have forgot about this when updating.

**Comment by sfackler** [comment]
Updated!

**Pull request merged**